internal/v2/certverifier/certverifier_test.go

```diff
@@ -167,28 +171,39 @@ func TestVerifyServerCertChain(t *testing.T) {
 	}
 
 	for _, tc := range []struct {
-		description string
-		hostname    string
-		rawCerts    [][]byte
-		expectedErr error
+		description               string
+		hostname                  string
+		rawCerts                  [][]byte
+		expectedErr               error
+		serverAuthorizationPolicy []byte
 	}{
 		{
-			description: "empty chain",
-			hostname:    "host",
-			rawCerts:    nil,
-			expectedErr: errors.New("server cert verification failed: server peer verification failed: server cert chain is empty"),
+			description:               "empty chain",
+			hostname:                  "host",
+			rawCerts:                  nil,
+			expectedErr:               errors.New("server cert verification failed: server peer verification failed: server cert chain is empty"),
+			serverAuthorizationPolicy: []byte(fakeServerAuthorizationPolicy),
 		},
 		{
-			description: "chain of length 1",
-			hostname:    "host",
-			rawCerts:    [][]byte{serverRootDERCert},
-			expectedErr: nil,
+			description:               "invalid server authorization policy",
+			hostname:                  "host",
+			rawCerts:                  [][]byte{serverRootDERCert},
+			expectedErr:               errors.New(fmt.Sprintf("rpc error: code = Unknown desc = server peer verification failed: invalid server authorization policy, expected: %s, got: .", fakeServerAuthorizationPolicy)),
+			serverAuthorizationPolicy: nil,
 		},
 		{
-			description: "chain of length 2 correct",
-			hostname:    "host",
-			rawCerts:    [][]byte{serverLeafDERCert, serverIntermediateDERCert},
-			expectedErr: nil,
+			description:               "chain of length 1",
```

Comment:
Should we have some cases where we succeed but the server authz policy is nil?

Reply:
Done.